File: sample/src/androidTest/java/com/andremion/counterfab/sample/MainActivityTest.java
package com.andremion.counterfab.sample;

import android.content.pm.ActivityInfo;

import com.facebook.testing.screenshot.Screenshot;

import org.junit.Rule;
import org.junit.Test;

import androidx.test.platform.app.InstrumentationRegistry;
import androidx.test.rule.ActivityTestRule;

import static androidx.test.espresso.Espresso.onView;
import static androidx.test.espresso.action.ViewActions.click;
import static androidx.test.espresso.matcher.ViewMatchers.withId;

public class MainActivityTest {

    @Rule
    public final ActivityTestRule<MainActivity> testRule = new ActivityTestRule<>(MainActivity.class);

    @Test
    public void shouldRenderProperly() {

        Screenshot.snapActivity(testRule.getActivity()).record();
    }

    @Test
    public void shouldRenderProperlyAfterOrientationChanged() {

        testRule.getActivity().setRequestedOrientation(ActivityInfo.SCREEN_ORIENTATION_LANDSCAPE);
        InstrumentationRegistry.getInstrumentation().waitForIdleSync();

        Screenshot.snapActivity(testRule.getActivity()).record();
    }

    @Test
    public void shouldKeepStateAfterOrientationChanged() {
        onView(withId(R.id.fab)).perform(click(), click());

        testRule.getActivity().setRequestedOrientation(ActivityInfo.SCREEN_ORIENTATION_LANDSCAPE);
        InstrumentationRegistry.getInstrumentation().waitForIdleSync();

        Screenshot.snapActivity(testRule.getActivity()).record();
    }
}

File: sample/src/androidTest/java/com/andremion/counterfab/sample/MainActivityTest.kt
package com.andremion.counterfab.sample

import android.content.pm.ActivityInfo
import androidx.test.espresso.Espresso.onView
import androidx.test.espresso.action.ViewActions.click
import androidx.test.espresso.matcher.ViewMatchers.withId
import androidx.test.platform.app.InstrumentationRegistry
import androidx.test.rule.ActivityTestRule
import com.andremion.counterfab.sample.helpers.setFabSize
import com.facebook.testing.screenshot.Screenshot
import com.google.android.material.floatingactionbutton.FloatingActionButton
import org.junit.Rule
import org.junit.Test

class MainActivityTest {

    @get:Rule
    val testRule = ActivityTestRule(MainActivity::class.java)

    @Test
    fun shouldRenderProperly() {

        Screenshot.snapActivity(testRule.activity).record()
    }

    @Test
    fun shouldRenderProperlyAfterOrientationChanged() {

        testRule.activity.requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_LANDSCAPE
        InstrumentationRegistry.getInstrumentation().waitForIdleSync()

        Screenshot.snapActivity(testRule.activity).record()
    }

    @Test
    fun shouldKeepStateAfterOrientationChanged() {
        onView(withId(R.id.fab)).perform(click(), click())

        testRule.activity.requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_LANDSCAPE
        InstrumentationRegistry.getInstrumentation().waitForIdleSync()

        Screenshot.snapActivity(testRule.activity).record()
    }

    @Test
    fun shouldRenderSizeMiniProperly() {
        onView(withId(R.id.fab)).perform(setFabSize(FloatingActionButton.SIZE_MINI))

        Screenshot.snapActivity(testRule.activity).record()
    }

    @Test
    fun shouldRenderSizeMiniProperlyAfterOrientationChanged() {
        onView(withId(R.id.fab)).perform(setFabSize(FloatingActionButton.SIZE_MINI))

        testRule.activity.requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_LANDSCAPE
        InstrumentationRegistry.getInstrumentation().waitForIdleSync()

        Screenshot.snapActivity(testRule.activity).record()
    }

    @Test
    fun shouldKeepSizeMiniStateAfterOrientationChanged() {
        onView(withId(R.id.fab)).perform(click(), click())

        testRule.activity.requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_LANDSCAPE
        InstrumentationRegistry.getInstrumentation().waitForIdleSync()

        Screenshot.snapActivity(testRule.activity).record()
    }
}